debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlConnection.java

```diff
@@ -0,0 +1,551 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.debezium.connector.mysql;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.debezium.DebeziumException;
+import io.debezium.config.CommonConnectorConfig;
+import io.debezium.config.CommonConnectorConfig.EventProcessingFailureHandlingMode;
+import io.debezium.config.Configuration;
+import io.debezium.config.Configuration.Builder;
+import io.debezium.config.Field;
+import io.debezium.connector.mysql.MySqlConnectorConfig.SecureConnectionMode;
+import io.debezium.connector.mysql.legacy.MySqlJdbcContext.DatabaseLocales;
+import io.debezium.jdbc.JdbcConnection;
+import io.debezium.relational.TableId;
+import io.debezium.relational.history.DatabaseHistory;
+import io.debezium.util.Strings;
+
+/**
+ * {@link JdbcConnection} extension to be used with MySQL Server
+ *
+ * @author Jiri Pechanec, Randall Hauch
+ *
+ */
+public class MySqlConnection extends JdbcConnection {
```

Comment:
Where is this coming from, didn't we have it before?

Reply:
No, we did not, there si {io.debezium.connector.mysql.legacy.MySqlJdbcContext}. I refactored the code and created a unified MySqlConnection to be same as with other connectors.